src/wallet/wallet.cpp

```diff
@@ -1979,7 +1984,9 @@ void CWallet::ResubmitWalletTransactions(bool relay, bool force)
 void MaybeResendWalletTxs(WalletContext& context)
 {
     for (const std::shared_ptr<CWallet>& pwallet : GetWallets(context)) {
+        if (!pwallet->ShouldResend()) continue;
         pwallet->ResubmitWalletTransactions(/*relay=*/true, /*force=*/false);
+        pwallet->SetNextResend();
```

Comment:
Might be good to have bugfixes in separate commits from refactoring changes

Reply:
The "bugfix" commit is full of refactoring still

Reply:
Also, cherry-picking the bugfix commit by itself is broken: it will reset `m_next_resend` to 12-36 hours, *every minute*, even if nothing is done/rebroadcast.